volatility/framework/plugins/linux/check_idt.py

```diff
@@ -0,0 +1,94 @@
+# This file is Copyright 2019 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+from typing import List
+
+from volatility.framework import interfaces, renderers, constants, contexts, exceptions, symbols
+from volatility.framework.configuration import requirements
+from volatility.framework.renderers import format_hints
+from volatility.framework.symbols import linux
+from volatility.plugins.linux import lsmod
+
+vollog = logging.getLogger(__name__)
+
+
+class Check_idt(interfaces.plugins.PluginInterface):
+    """ Checks if the IDT has been altered """
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.TranslationLayerRequirement(name='primary',
+                                                     description='Memory layer for the kernel',
+                                                     architectures=["Intel32", "Intel64"]),
+
+            requirements.SymbolTableRequirement(name="vmlinux", description="Linux kernel symbols"),
+            
+            requirements.PluginRequirement(name='lsmod', plugin=lsmod.Lsmod, version=(1, 0, 0))
+        ]
+
+    def _generator(self):
+        vmlinux = contexts.Module(self.context, self.config['vmlinux'], self.config['primary'], 0)
+
+        modules = lsmod.Lsmod.list_modules(self.context, self.config['primary'], self.config['vmlinux'])
+
+        handlers = linux.LinuxUtilities.generate_kernel_handler_info(self.context, self.config['primary'], self.config['vmlinux'], modules)
+
+
+        is_32bit = not symbols.symbol_table_is_64bit(self.context, self.config["vmlinux"])
+
+        tblsz = 256
```

Comment:
As long as this variable name is clear/obvious in the world of linux, this is ok.  If it's just shorthand, then long hand is more descriptive...  5;)

Reply:
this is directly copied from vol2 so Im not sure how clear and obvious it would be. Happy to change it if you feel it would make it more readable

Reply:
I'm happy for @atcuno to weigh in, but the fact that's how vol2 did it isn't enough reason to keep it that way if there's reason to change it (clarity and readability is a goal of vol3).  As I say, if it immediately tells linux people what they expect, that's fine, if it's *extreme* effort saving by removing vowels, then we should probably expand on it a bit...  5;)

Reply:
I would agree. Based on a quick google their are 256 entries in a idt table so it would seem like it refers to that. Changing it to something like idt_table_size seems like a better name

Reply:
YES!  That is *so* much more descriptive than tblsz (although, looking at it now I can assume it was a shortening of tablesize, but before that it could've been anything!!!  Sleeping tables, tomatoes blowsz, someone sneezing as they typed the variable name).  So yes, `idt_table_size` would be a vast improvement!!!  5:D